Automation resources

bgelens opened the pull request:
New resources for Automation Services

* azurerm_automation_module
* azurerm_automation_dsc_configuration
* azurerm_automation_dsc_nodeconfiguration

I am new to almost every aspect of this (including development in go) so if it's not that "good" I totally understand but I'm seeking to learn!

Technically, dsc_nodeconfiguration has a hard dependency on dsc_configuration. The portal creates an empty configuration using special magic but the API doesn't expose this coupling of actions.

lfshr commented:
The dsc_configuration stuff should be relatively simple. The azurerm_automation_module may be a different beast depending on what you want to do with it. You have to supply a blob location to a zip file for it to work, you can't supply local files. There was an issue open on one of their repos for it, I'll try and find it. You could add an azurerm_automation_module resource but I would probably follow the same principle (pass it a blob location). Which you  could upload using the azurerm_storage_blob.

lfshr commented:
I totally didn't realise this was a PR and not an issue. Sorry bud, my last comment probably didn't make any sense.

lfshr commented on azurerm/resource_arm_automation_dsc_configuration.go:
```diff
@@ -0,0 +1,141 @@
+				ForceNew: true,
+			},
+
+			"account_name": {
```
I would probably be explicit here and call this automation_account_name

katbyte replied:
Agreed, this should be `automation_account_name`

bgelens replied:
I was following already established Automation related resources like [credential](https://www.terraform.io/docs/providers/azurerm/r/automation_credential.html)

```tf
resource "azurerm_automation_credential" "example" {
  name                = "credential1"
  resource_group_name = "${azurerm_resource_group.example.name}"
  account_name        = "${azurerm_automation_account.example.name}"
  username           = "example_user"
  password            = "example_pwd"
  description         = "This is an example credential"
}
```

Do I still need to change mine (it will result in inconsistent parameter naming)?

lfshr replied:
Ah. Makes me slightly sad but I'd probably say consistency over correctness :cry: @katbyte your call.

lfshr replied:
Or we could deprecate the account_name in azurerm_automation_credential.

katbyte replied:
Please do, the other resources are incorrect and will need to be changed eventually.

As much as I would usually agree in consistency over correctness, this isn't a very confusing change so for a new resource I'm going to go with its better to use the correct name.

I have already depreciated `account_name` in `azurerm_automation_schedule`, so feel free to apply that to the rest of the automation resources, preferably in a separate PR 🙂

bgelens commented:
No worries! I figured as much.
I'm going on holiday without internet for the coming week so take your time for reviewing.

lfshr commented on azurerm/resource_arm_automation_module.go:
```diff
@@ -0,0 +1,186 @@
+				ForceNew: true,
+			},
+
+			"account_name": {
```
Same as above

lfshr commented on azurerm/resource_arm_automation_dsc_nodeconfiguration.go:
```diff
@@ -0,0 +1,141 @@
+				ForceNew: true,
+			},
+
+			"account_name": {
```
Same as above

katbyte replied:
Agreed

lfshr commented:
I won't be doing the actual review, I'm really on the same boat as yourself re. GO. Only the one comment from my untrained eye. 😄

katbyte commented on azurerm/config.go:
```diff
@@ -71,10 +71,13 @@ type ArmClient struct {
+	automationCredentialClient           automation.CredentialClient
+	automationScheduleClient             automation.ScheduleClient
+	automationModuleClient               automation.ModuleClient
+	automationDscNodeConfigurationClient automation.DscNodeConfigurationClient
```
Could we get these in alphabetical order?

katbyte commented on azurerm/provider.go:
```diff
@@ -128,6 +128,9 @@ func Provider() terraform.ResourceProvider {
 			"azurerm_automation_schedule":                  resourceArmAutomationSchedule(),
+			"azurerm_automation_module":                    resourceArmAutomationModule(),
+			"azurerm_automation_dsc_nodeconfiguration":     resourceArmAutomationDscNodeConfiguration(),
+			"azurerm_automation_dsc_configuration":         resourceArmAutomationDscConfiguration(),
```
As above, could we put these into alphabetical order?

katbyte commented on azurerm/resource_arm_automation_dsc_configuration.go:
```diff
@@ -0,0 +1,141 @@
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
```
It would be nice to have some validation on the name here, but not required.

Example [here](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/resource_arm_automation_schedule.go#L35)

katbyte commented on azurerm/resource_arm_automation_dsc_configuration.go:
```diff
@@ -0,0 +1,141 @@
+
+			"content": {
+				Type:     schema.TypeString,
+				Required: true,
```
Unless an empty string is valid, a quick `validation.NoZeroValues` would be nice to have here.

katbyte commented on azurerm/resource_arm_automation_dsc_configuration.go:
```diff
@@ -0,0 +1,141 @@
+
+			"location": locationSchema(),
+
+			"content": {
```
As this is actually embedded content, could we call this property `content_embedded` so `content_uri` could be used for the other type?

katbyte commented on azurerm/resource_arm_automation_dsc_configuration.go:
```diff
@@ -0,0 +1,141 @@
+
+		return fmt.Errorf("Error making Read request on AzureRM Automation Dsc Configuration '%s': %+v", name, err)
+	}
+
```
Is there a reason you are not fetching the content? It does appear to be part of the data structure. If there is a reason a comment explaining why here would be great.

katbyte replied:
Also is there a reason location not being read back into state?

bgelens replied:
Content is not part of the data structure, instead it needs to be fetched by GetContent function. I have an issue with the Azure Go SDK so I won't be able to fix this right now. Reported an [issue](https://github.com/Azure/azure-sdk-for-go/issues/2486)

katbyte replied:
Looks like that issue has been resolved

katbyte commented on azurerm/resource_arm_automation_dsc_configuration_test.go:
```diff
@@ -0,0 +1,130 @@
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"content", "location"},
```
A comment as to why these are being ignore would be good

katbyte commented on azurerm/resource_arm_automation_dsc_configuration_test.go:
```diff
@@ -0,0 +1,130 @@
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  account_name        = "${azurerm_automation_account.test.name}"
+  location            = "${azurerm_resource_group.test.location}"
+	content             = "configuration test {}"
```
there is an extra space here infront of content

katbyte commented on azurerm/resource_arm_automation_dsc_configuration_test.go:
```diff
@@ -0,0 +1,130 @@
+func testAccAzureRMAutomationDscConfiguration_basic(rInt int, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+ name = "acctestRG-%d"
```
We should be indenting with two spaces here

katbyte commented on azurerm/resource_arm_automation_dsc_configuration_test.go:
```diff
@@ -0,0 +1,130 @@
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  sku {
+	name = "Basic"
```
This should be indent with 4 spaces

katbyte commented on azurerm/resource_arm_automation_dsc_configuration_test.go:
```diff
@@ -0,0 +1,130 @@
+			{
+				Config: testAccAzureRMAutomationDscConfiguration_basic(ri, testLocation()),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMAutomationDscConfigurationExists(resourceName),
```
We should be checking the values returned and read into state as can be see [here](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/resource_arm_scheduler_job_test.go#L57)

bgelens replied:
done

katbyte commented on azurerm/resource_arm_automation_dsc_nodeconfiguration.go:
```diff
@@ -0,0 +1,141 @@
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"content": {
```
As above, i think this should be `content_embedded`

katbyte commented on azurerm/resource_arm_automation_dsc_nodeconfiguration.go:
```diff
@@ -0,0 +1,141 @@
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resGroup)
+	d.Set("account_name", accName)
+
```
Again, is there a reason location and content are not being read back and set into state? If so please leave a comment.

katbyte commented on azurerm/resource_arm_automation_dsc_nodeconfiguration.go:
```diff
@@ -0,0 +1,141 @@
+				Required: true,
+				ForceNew: true,
+			},
+
```
Is there a reason location was left off this resource?

katbyte commented on azurerm/resource_arm_automation_dsc_nodeconfiguration_test.go:
```diff
@@ -0,0 +1,160 @@
+			{
+				Config: testAccAzureRMAutomationDscNodeConfiguration_basic(ri, testLocation()),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMAutomationDscNodeConfigurationExists(resourceName),
```
Again we should be checking the state of the resource not just that it exists.

bgelens replied:
Added resource attribute check for configuration_name

katbyte commented on azurerm/resource_arm_automation_dsc_nodeconfiguration_test.go:
```diff
@@ -0,0 +1,160 @@
+
+func testAccAzureRMAutomationDscNodeConfiguration_basic(rInt int, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
```
This terraform should be properly formatted, see my comments above

bgelens replied:
done

katbyte commented on azurerm/resource_arm_automation_module.go:
```diff
@@ -0,0 +1,186 @@
+
+		return fmt.Errorf("Error making Read request on AzureRM Automation Module '%s': %+v", name, err)
+	}
+
```
Is there a reason `module_link` is not being read back into state here?

katbyte commented on azurerm/resource_arm_automation_module_test.go:
```diff
@@ -0,0 +1,132 @@
+			{
+				Config: testAccAzureRMAutomationModule_basic(ri, testLocation()),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMAutomationModuleExists(resourceName),
```
Again, we should be checking the actual state of the resource after a apply not just if it exists.

bgelens commented:
I'm back from holiday.. sorry for the delayed response 😄 I will start working on the issues mentioned!

bgelens commented:
Sorry I'm not making great progress. Working on it when I can. I'll request a new review once I have gone through all remarks (hopefully end of week)

JunyiYi commented on azurerm/resource_arm_automation_module.go:
```diff
@@ -0,0 +1,186 @@
+			"module_link": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
```
>MaxItems: 1, [](start = 4, length = 12)

*suggestion* Since this object will occur at most once and the whole resource represents a module, will it be better to remove `module_link` object and just move `uri` and `hash` to the top level named as `module_uri` and `module_hash`?

bgelens replied:
I used the same implementation as with the runbook resource.

JunyiYi commented on azurerm/resource_arm_automation_module.go:
```diff
@@ -0,0 +1,186 @@
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read Automation Module '%s' (resource group %s) ID", name, resGroup)
```
>%s [](start = 52, length = 2)

We use `%q` instead of `%s` for name and resource group name.

bgelens replied:
I copied this from the other automation resources which where already part of the provider. Changed the resources in scope of PR but fyi, the others do still have %s

JunyiYi commented on azurerm/resource_arm_automation_module.go:
```diff
@@ -0,0 +1,186 @@
+			return nil
+		}
+
+		return fmt.Errorf("Error issuing AzureRM delete request for Automation Module '%s': %+v", name, err)
```
*minor* Maybe reverse the condition here for clearer logic:

```go
if err != nil {
  if !utils.ResponseWasNotFound(resp) {
    return fmt.Errorf(...)
  }
}
```

bgelens replied:
I copied this from the other automation resources. Should we drift from these or clean them all up in one go through another PR?

JunyiYi commented on azurerm/resource_arm_automation_module.go:
```diff
@@ -0,0 +1,186 @@
+func expandModuleLink(d *schema.ResourceData) automation.ContentLink {
+	inputs := d.Get("module_link").([]interface{})
+	input := inputs[0].(map[string]interface{})
+	uri := input["uri"].(string)
```
*sidenote* We have a shortcut here: `uri := d.Get("module_link.0.uri").(string)`.

JunyiYi commented on azurerm/resource_arm_automation_module_test.go:
```diff
@@ -0,0 +1,132 @@
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"module_link"},
```
>ImportStateVerifyIgnore: []string{"module_link"}, [](start = 4, length = 49)

Why not import the `module_link` here? This should be the similar comment provided in `resource_arm_automation_module.go`.

bgelens replied:
the content link is not returned by the rest api in a get operation.  This is the body I get returned:
```json
{
  "id": "/subscriptions/xxx/resourceGroups/xxx/providers/Microsoft.Automation/automationAccounts/xxxx/modules/xActiveDirectory",
  "name": "xActiveDirectory",
  "type": "Microsoft.Automation/AutomationAccounts/Modules",
  "location": "westeurope",
  "tags": {},
  "etag": null,
  "properties": {
    "isGlobal": false,
    "version": "2.19.0.0",
    "sizeInBytes": 101851,
    "activityCount": 13,
    "creationTime": "2018-08-20T06:32:33.79+00:00",
    "lastModifiedTime": "2018-08-20T06:33:45.133+00:00",
    "error": {
      "code": null,
      "message": ""
    },
    "provisioningState": "Succeeded",
    "isComposite": false
  }
}
```

JunyiYi commented on azurerm/resource_arm_automation_dsc_configuration.go:
```diff
@@ -0,0 +1,141 @@
+			return nil
+		}
+
+		return fmt.Errorf("Error issuing AzureRM delete request for Automation Configuration '%s': %+v", name, err)
```
>Automation Configuration [](start = 62, length = 24)

*minor* Automation DSC Configuration

JunyiYi commented on azurerm/resource_arm_automation_dsc_nodeconfiguration.go:
```diff
@@ -0,0 +1,141 @@
+	accName := d.Get("automation_account_name").(string)
+	content := d.Get("content").(string)
+
+	s := strings.Split(name, ".")
```
Why are you splitting the `name`? It seems that you are not using `s[1]` in the code below.

JunyiYi commented on azurerm/resource_arm_automation_dsc_nodeconfiguration.go:
```diff
@@ -0,0 +1,141 @@
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resGroup)
+	d.Set("automation_account_name", accName)
```
Are there any specific reason not setting back `content` here?

bgelens replied:
It's not part of the returned properties. There is no way to fetch it afaik.
When looking at the [REST API specs](https://docs.microsoft.com/en-us/rest/api/automation/dscnodeconfiguration/get), it just is not there.

JunyiYi commented on website/docs/r/automation_module.html.markdown:
```diff
@@ -0,0 +1,61 @@
+---
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_automation_module"
+sidebar_current: "docs-azurerm-resource-automation-module"
```
>docs-azurerm-resource-automation-module [](start = 18, length = 39)

You need to add this page to the ToC in [`azurerm.erb`](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/website/azurerm.erb) as well.

bgelens replied:
done

bgelens commented on azurerm/resource_arm_automation_module.go:
```diff
@@ -0,0 +1,186 @@
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
```
Done

bgelens commented:
@JunyiYi @katbyte I think I addressed most concerns. I'm struggling with the Azure GO SDK a bit as it promises to return a lot of things which are actually not returned by the automation RP.

I have an outstanding [issue](https://github.com/Azure/azure-sdk-for-go/issues/2486) for the SDK where it implements GetContent for the DSC Configuration. I've commented the code that at this time we cannot fetch the content due to this issue.

Looking forward to your feedback!

lfshr commented:
@bgelens I'd be hesitant to include it in this PR as it's starting to look pretty good now, but should we think of adding the ability to compile the DSC configurations via terraform? (I had a look and couldn't see this anywhere)

bgelens commented:
@lfshr That would be another [resource](https://github.com/Azure/azure-sdk-for-go/blob/master/services/automation/mgmt/2015-10-31/automation/dsccompilationjob.go) to create a dsc compilation job and it should also monitor the job. The resources in this PR are needed to at least handle a pre-compiled mof.

I would prefer not to include this extra resource in this PR (looking back at this one, I would be better off handling 1 per PR to keep it a bit simpler).

lfshr commented:
@bgelens no arguments from me re. the inclusion of the resource in this PR. I can have a Go at it (see what I did there?) Or I can leave it to you. Be nice to also include the ability to request modules from the gallery. I have some PowerShell code that I can translate to go that achieves it. What do you think?

JunyiYi commented on azurerm/resource_arm_automation_dsc_configuration.go:
```diff
@@ -0,0 +1,183 @@
+
+			"content_embedded": {
+				Type:         schema.TypeString,
+				Required:     true,
```
*minor* Convention: Typically we sort the fields in "Required -> Optional -> Computed" order.

bgelens replied:
Done

JunyiYi commented on azurerm/resource_arm_automation_dsc_configuration.go:
```diff
@@ -0,0 +1,183 @@
+
+	parameters := automation.DscConfigurationCreateOrUpdateParameters{
+		DscConfigurationCreateOrUpdateProperties: &automation.DscConfigurationCreateOrUpdateProperties{
+			LogVerbose:  &logVerbose,
```
*minor* Latest coding style: Always using `utils.Bool`/`utls.String` instead of address of `&` for primitive types.

bgelens commented:
@JunyiYi I've fixed your minor comments. I'll need to figure out what to do to reference the newer SDK once it is out so I can include the content fetch. Any pointers here would be greatly appreciated :)

@lfshr Please Go ahead creating that compilation resource! With regard to the psgallery translation, you think to embed that logic in the module resource? That would be a great addition IMO

lfshr commented:
@bgelens see [govendor](https://github.com/kardianos/govendor) for help on updating the vendor packages.

I'm not sure. I think I'll keep it the same as the swagger and maybe think about including it in the dsc_configuration as well.

bgelens commented:
@JunyiYi I've updated to [v20.0.0](https://github.com/Azure/azure-sdk-for-go/releases/tag/v20.0.0) for automation and added the content fetch for the dsc configuration resource (updated test accordingly).

bgelens commented on azurerm/resource_arm_automation_module.go:
```diff
@@ -0,0 +1,189 @@
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resGroup)
+	d.Set("automation_account_name", accName)
+
```
It is not part of the object state in Azure Automation account so it cannot be read back.

```json
{
  "id": "/subscriptions/xxxx/resourceGroups/bla/providers/Microsoft.Automation/automationAccounts/mytfaaaccount/modules/xActiveDirectory",
  "name": "xActiveDirectory",
  "type": "Microsoft.Automation/AutomationAccounts/Modules",
  "location": "westeurope",
  "tags": {},
  "etag": null,
  "properties": {
    "isGlobal": false,
    "version": "2.19.0.0",
    "sizeInBytes": 101851,
    "activityCount": 13,
    "creationTime": "2018-08-20T08:32:33.79+02:00",
    "lastModifiedTime": "2018-08-24T07:56:43.897+02:00",
    "error": {
      "code": null,
      "message": ""
    },
    "provisioningState": "Succeeded",
    "isComposite": false
  }
}
```

bgelens commented on azurerm/resource_arm_automation_dsc_configuration_test.go:
```diff
@@ -0,0 +1,135 @@
+}
+
+resource "azurerm_automation_dsc_configuration" "test" {
+  name                    = "test"
```
This is already the case for the automation account. Resources in scope of this pr cannot exist without the automation account so there is no need to change this

katbyte replied:
Hi @bgelens, 

We name all resources in a acceptance test `acctest`. As there is no reason not to in this case it would be great if we could change it to align with our testing conventions.

bgelens commented on azurerm/resource_arm_automation_dsc_nodeconfiguration_test.go:
```diff
@@ -0,0 +1,162 @@
+}
+
+resource "azurerm_automation_dsc_configuration" "test" {
+  name                    = "test"
```
Same as previous reply. No need to do this

katbyte replied:
As mentioned above, we would prefer this to be name `acctest`.

bgelens commented on azurerm/resource_arm_automation_dsc_configuration.go:
```diff
@@ -0,0 +1,191 @@
+			},
+		},
+		Location: &location,
+		Name:     &name,
```
It will fail [validation](https://github.com/Azure/azure-sdk-for-go/blob/master/services/automation/mgmt/2015-10-31/automation/dscnodeconfiguration.go#L62).

bgelens commented on azurerm/resource_arm_automation_dsc_nodeconfiguration.go:
```diff
@@ -0,0 +1,156 @@
+		Configuration: &automation.DscConfigurationAssociationProperty{
+			Name: &configurationName,
+		},
+		Name: &name,
```
It will fail [validation](https://github.com/Azure/azure-sdk-for-go/blob/master/services/automation/mgmt/2015-10-31/automation/dscnodeconfiguration.go#L62).

bgelens commented:
@JunyiYi ```why not set back content_embedded and module_link in DSC Node Configuration and Automation Module?``` Unfortunately, getting the content_embedded for DSC Node configuration is not exposed via the API, nor is the module link for Automation module.

bgelens commented:
I changed the resource name to acctest for both test cases. No dashes are allowed so I hope this is ok like this.

bgelens commented:
Wondering if there is anything else to fix or if it is ok like this? Thanks!

katbyte commented:
Tests pass:
```
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./azurerm -v -test.run=TestAccAzureRMAutoma -timeout 180m -ldflags="-X=github.com/terraform-providers/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccAzureRMAutomationAccount_basic
--- PASS: TestAccAzureRMAutomationAccount_basic (75.32s)
=== RUN   TestAccAzureRMAutomationAccount_complete
--- PASS: TestAccAzureRMAutomationAccount_complete (70.27s)
=== RUN   TestAccAzureRMAutomationCredential_basic
--- PASS: TestAccAzureRMAutomationCredential_basic (71.22s)
=== RUN   TestAccAzureRMAutomationCredential_complete
--- PASS: TestAccAzureRMAutomationCredential_complete (70.70s)
=== RUN   TestAccAzureRMAutomationDscConfiguration_basic
--- PASS: TestAccAzureRMAutomationDscConfiguration_basic (74.23s)
=== RUN   TestAccAzureRMAutomationDscNodeConfiguration_basic
--- PASS: TestAccAzureRMAutomationDscNodeConfiguration_basic (77.86s)
=== RUN   TestAccAzureRMAutomationModule_basic
--- PASS: TestAccAzureRMAutomationModule_basic (73.08s)
=== RUN   TestAccAzureRMAutomationRunbook_PSWorkflow
--- PASS: TestAccAzureRMAutomationRunbook_PSWorkflow (73.43s)
=== RUN   TestAccAzureRMAutomationRunbook_PSWorkflowWithHash
--- PASS: TestAccAzureRMAutomationRunbook_PSWorkflowWithHash (73.03s)
=== RUN   TestAccAzureRMAutomationRunbook_PSWithContent
--- PASS: TestAccAzureRMAutomationRunbook_PSWithContent (76.68s)
=== RUN   TestAccAzureRMAutomationSchedule_oneTime_basic
--- PASS: TestAccAzureRMAutomationSchedule_oneTime_basic (74.96s)
=== RUN   TestAccAzureRMAutomationSchedule_oneTime_complete
--- PASS: TestAccAzureRMAutomationSchedule_oneTime_complete (70.93s)
=== RUN   TestAccAzureRMAutomationSchedule_oneTime_update
--- PASS: TestAccAzureRMAutomationSchedule_oneTime_update (81.99s)
=== RUN   TestAccAzureRMAutomationSchedule_hourly
--- PASS: TestAccAzureRMAutomationSchedule_hourly (70.93s)
=== RUN   TestAccAzureRMAutomationSchedule_daily
--- PASS: TestAccAzureRMAutomationSchedule_daily (71.28s)
=== RUN   TestAccAzureRMAutomationSchedule_weekly
--- PASS: TestAccAzureRMAutomationSchedule_weekly (70.92s)
=== RUN   TestAccAzureRMAutomationSchedule_monthly
--- PASS: TestAccAzureRMAutomationSchedule_monthly (70.87s)
=== RUN   TestAccAzureRMAutomationSchedule_weekly_advanced
--- PASS: TestAccAzureRMAutomationSchedule_weekly_advanced (70.76s)
=== RUN   TestAccAzureRMAutomationSchedule_monthly_advanced_by_day
--- PASS: TestAccAzureRMAutomationSchedule_monthly_advanced_by_day (69.89s)
=== RUN   TestAccAzureRMAutomationSchedule_monthly_advanced_by_week_day
--- PASS: TestAccAzureRMAutomationSchedule_monthly_advanced_by_week_day (69.83s)
PASS
ok  	github.com/terraform-providers/terraform-provider-azurerm/azurerm	1458.224s
```
